base/abstractdict.jl

```diff
@@ -579,6 +579,55 @@ _tablesz(x::T) where T <: Integer = x < 16 ? T(16) : one(T)<<(top_set_bit(x-one(
 
 TP{K,V} = Union{Type{Tuple{K,V}},Type{Pair{K,V}}}
 
+# validate the contents of this iterator by testing the iteration protocol (isiterable) on it
+_throw_dict_kv_error() = throw(ArgumentError("AbstractDict(kv): kv needs to be an iterator of 2-tuples or pairs"))
```

Comment:
The comment looks as if it is attached to `_throw_dict_kv_error()` but it seems to talk about something else?

Reply:
Good point. This is the error thrown when the check described in that comment fails. The comment and function used to be inline in the function, and then I moved both here to avoid repeating them every time it called the function

Reply:
```suggestion
# This error is throw if `grow_to!` cannot validate the contents of the iterator argument, which it does by testing the iteration protocol (isiterable) on it each time it is about to start iteration on it
_throw_dict_kv_error() = throw(ArgumentError("AbstractDict(kv): kv needs to be an iterator of 2-tuples or pairs"))
```